source/_app/pages/scheduled.js

```diff
@@ -29,10 +50,110 @@ app.scheduled = {
   },
 
   /**
+   * Generate the url for history.js
+   * @returns {string}
+   */
+  createUrl: function() {
+
+    // put the URL together
+    var url = '?';
+    if (this.currentView) {
+      url += 'view=' + this.currentView;
+
+      if (this.currentView ==  'list') {
+        if (this.listDateStart) {
+          url += '&start=' + this.listDateStart;
+        }
+
+        if (this.listDateEnd) {
+          url += '&end=' + this.listDateEnd;
+        }
+      }
+
+      if (this.currentView ==  'calendar') {
+        if (this.currentViewType) {
+          url += '&type=' + this.currentViewType;
+        }
+
+        if (this.calendarDate) {
+          url += '&date=' + this.calendarDate;
+        }
+      }
+    }
+
+    return url;
+  },
+
+  getUrlParams: function () {
+    var state = History.getState();
+    var hash = state.hash;
+    var urlObject = app.utils.urlObject(hash);
+    return urlObject.parameters;
+  },
+
+  /**
+   * Set the page url when first loading the page
+   */
+  setPageUrl: function() {
+    // We're setting the page url here so the priority is to take items from the URL, else we will use the defaults from init
+    this.urlParams = this.getUrlParams();
+
+
+    // set from url
+    if (_.has(this.urlParams, 'view')) {
+      this.currentView = this.urlParams.view;
+    }
+
+    if (_.has(this.urlParams, 'start')) {
+      this.listDateStart = this.urlParams.start;
+    }
+
+    if (_.has(this.urlParams, 'end')) {
+      this.listDateEnd = this.urlParams.end;
+    }
+
+    if (_.has(this.urlParams, 'type')) {
+      this.currentViewType = this.urlParams.type;
+    }
+
+    if (_.has(this.urlParams, 'date')) {
+      this.calendarDate = this.urlParams.date;
+    }
+
+    var url = this.createUrl();
+    History.replaceState(null, null, url);
+  },
+
+  /**
+   * Update the page url when things happen. Things such as, clicking list/calendar button or changing the view in the calendar
+   */
+  updatePageUrl: function() {
+    // here we have updated the globals so we will take the url data from the history object
+    // unless its the list view where we use the default dates
+    // (if we were coming from the updated url ones the setPageUrl  method would trigger instead of this one)
+    if (this.currentView == 'list') {
+      this.listDateStart = this.defaultlistDateStart;
+      this.listDateEnd = this.defaultlistDateEnd;
+    }
+
+    var url = this.createUrl();
+    History.pushState(null, null, url);
+  },
+
+  /**
    * Bind events
    */
   bindEvents: function() {
     $(this.$el).on('click', '.schedule-page-switch', this.clickSwitchPage.bind(this));
+    $(window).on('statechange', this.stateChange.bind(this));
+  },
+
+  /**
+   * On back/forwards update the view
+   * @param e
+   */
+  stateChange: function(e) {
+    // this.switchPage(this.currentView);
```

Comment:
What should this do? Can it be deleted? Does the method need to exist as it might be called, but it should be a noop? Something else? If it needs to exist with an empty function body, please add a comment explaining.